### PR TITLE
[New Model]: Shared PartSiteInformationAsBuilt 1.0.0 

### LINKED ISSUE #442: [New Model]: Shared PartSiteInformationAsBuilt 1.0.0
## Model Description
Based on the results / definition of the Industry Core, the part site information is also needed for DTasBuilt. Therefore the existing aspect model PartSiteInformationAsPlanned will be used to fullfill the new requirements. The changes contain the implementation of a new shared aspect model as renamed version of the existing model and integrate this in the Industry Core aspect models SerialPart, Batch and JiS. 

The current PartSiteInformationAsPlanned model will be set to deprecated (see #443) after the current information were added to the PartAsPlanned model (#466).

## Relevant Standards
New Traceability Use Case Standard -> In Progress (#R24.03)

## Example Data
```json
{
  "catenaXId": "580d3adf-1981-44a0-a214-13d6ceed9379",
  "partSitesInformation": [
    {
      "catenaXsiteId": "BPNS1234567890ZZ",
      "function": "production"
    }
  ]
}
```

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by johannsvarela**
Based on the results / definition of the Industry Core, the part site information is also needed in the DTasBuilt context. Therefore a new aspect model was created, which contain the information of the sites of a part. The new model will be implemented as shared aspect model who will be integrated into the existing models SerialPart, Batch and JiS in further steps.

Closes #442

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by johannsvarela** [comment]
FYI @jonbckr @BenediktMuellerIML @jacewski-bosch

**Comment by johannsvarela** [comment]
@agg3fe there is an issue with the semantic model validation. Do you know what's the problem?

**Comment by agg3fe** [comment]
@johannsvarela do you want to add 'PartSiteInformationAsPlanned.ttl' also? If yes, then you have to create a separate Pull Request for this. we can have only one model (.ttl) file in one PR for new model.

**Comment by johannsvarela** [comment]
> @johannsvarela do you want to add 'PartSiteInformationAsPlanned.ttl' also? If yes, then you have to create a separate Pull Request for this. we can have only one model (.ttl) file in one PR for new model.

@agg3fe The second ttl-File (PartSiteInformationAsPlanned) contain the migration from BAMM to SAMM for the model. Since the PartSiteInformationAsPlanned will be set to 'deprecate', i don't know if its needed to migrate to SAMM.

Otherwise i will integrate the migration to the deprecation PR #454. Would it be fine for you?

**Comment by agg3fe** [comment]
> > @johannsvarela do you want to add 'PartSiteInformationAsPlanned.ttl' also? If yes, then you have to create a separate Pull Request for this. we can have only one model (.ttl) file in one PR for new model.
> 
> @agg3fe The second ttl-File (PartSiteInformationAsPlanned) contain the migration from BAMM to SAMM for the model. Since the PartSiteInformationAsPlanned will be set to 'deprecate', i don't know if its necessary to migrate to SAMM?
> 
> Otherwise i will integrate the SAMM-migration to the existing deprecation PR #454. Would it be fine for you?

If it is deprecated, then you don't need to migrate it from BAMM to SAMM. The separate PR is fine for me to set it to deprecate. you can remove the file from PR.

File: io.catenax.shared.part_site_information_as_built/1.0.0/PartSiteInformationAsBuilt.ttl [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+:SitesEntity a samm:Entity;
+   samm:preferredName "Site Entity"@en;
+   samm:description "Site Entity"@en;
+   samm:properties(:catenaXsiteId :function).
+
+:catenaXsiteId a samm:Property;
+   samm:preferredName "Catena-X site identifier"@en;
+   samm:description "The identifier of the site according to Catena-X BPDM. The catenaXsiteId must be a valid Catena-X BPN. The BPN is a unique, unchangeable identifier for Business Partners / company locations from foundation to closure, regardless of the different business relationships / structures between or within the Business Partners or company locations."@en;
+   samm:characteristic :BpnCharacteristic;
```
Comment (agg3fe) [inline io.catenax.shared.part_site_information_as_built/1.0.0/PartSiteInformationAsBuilt.ttl:68]: You can make use of existing characteristic and don't need to define a new one.
For reference: https://github.com/eclipse-tractusx/sldt-semantic-models/pull/405/files#diff-8294717bc9be9d276f548ae9f83e2831503589ef17d4d5bf6f1463cdb397d00eR104.
Just chose one correct for you from different type of BPN types in shared model.

File: io.catenax.shared.part_site_information_as_built/1.0.0/PartSiteInformationAsBuilt.ttl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   samm:characteristic ext-uuid:UuidV4Trait;
+   samm:exampleValue "580d3adf-1981-44a0-a214-13d6ceed9379".
+
+:SitesCharacteristic a samm-c:Set;
+   samm:preferredName "Sites Characteristic"@en;
+   samm:description "Sites Characteristic"@en;
+   samm:dataType :SitesEntity.
+
+:SitesEntity a samm:Entity;
+   samm:preferredName "Site Entity"@en;
```
Comment (agg3fe) [inline io.catenax.shared.part_site_information_as_built/1.0.0/PartSiteInformationAsBuilt.ttl:61]: preferredName and description should be different. please elaborate description a bit.

File: io.catenax.shared.part_site_information_as_built/1.0.0/PartSiteInformationAsBuilt.ttl [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+   samm:characteristic :FunctionCharacteristic;
+   samm:exampleValue "production".
+
+:BpnCharacteristic a samm:Characteristic;
+   samm:preferredName "BPN Characteristic"@en;
+   samm:description "A string of 16 characters:\nPrefix: 3 digits\t\t\t\t\nClassification: 1 digit\nIdentifier: 10 digits"@en;
+   samm:dataType xsd:string.
+
+:FunctionCharacteristic a samm-c:Enumeration;
+   samm:preferredName "Function Characteristic"@en;
```
Comment (agg3fe) [inline io.catenax.shared.part_site_information_as_built/1.0.0/PartSiteInformationAsBuilt.ttl:83]: please provide `description` also here

**Review by agg3fe — CHANGES_REQUESTED** [review]
Please check my comments

**Comment by johannsvarela** [comment]
> Please check my comments

@agg3fe The changes were implemented as you suggested! :-)

File: io.catenax.shared.part_site_information_as_built/RELEASE_NOTES.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+All notable changes to this model will be documented in this file.
+
+## [1.0.0] 2023-11-27
```
Comment (agg3fe) [inline io.catenax.shared.part_site_information_as_built/RELEASE_NOTES.md:5]: You can change the date here to 20th if you want MS3 approval today.

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by johannsvarela** [comment]
modeling_team

**Comment by johannsvarela** [comment]
@agg3fe Jira-Issue to generate the testdata set for this model was created: https://jira.catena-x.net/browse/TDG-35

**Pull request merged**